packages/async-flow/src/types.d.ts

```diff
@@ -71,8 +78,12 @@ export type GuestInterface<T> = {
  *
  * Specifically, Promise return values are converted to Vows.
  */
-export type HostOf<F> = F extends (...args: infer A) => Promise<infer R>
-  ? (...args: A) => Vow<R extends Passable ? R : HostInterface<R>>
+export type HostOf<F extends CallableFunction> = F extends (
+  ...args: infer A
+) => infer R
+  ? R extends Promise<infer T>
+    ? (...args: A) => Vow<T extends Passable ? T : HostInterface<T>>
+    : (...args: A) => HostInterface<R>
```

Comment:
If the return type is not passable, the host will never resolve

```suggestion
    ? (...args: A) => Vow<T extends Passable ? HostInterface<T> : never>
    : (...args: A) => R extends Passable ? HostInterface<R> : never
```